devito/passes/clusters/misc.py

```diff
@@ -166,79 +166,76 @@ class Key(tuple):
         two Clusters (ClusterGroups) are topo-fusible if and only if their Key is
         identical.
 
-        A Key contains several elements that can logically be split into two
-        groups -- the `strict` and the `weak` components of the Key.
-        Two Clusters (ClusterGroups) having same `strict` but different `weak` parts
-        are, as by definition, not fusible; however, since at least their `strict`
-        parts match, they can at least be topologically reordered.
+        A Key contains elements that can logically be split into two groups -- the
+        `strict` and the `weak` components of the Key. Two Clusters (ClusterGroups)
+        having same `strict` but different `weak` parts are, by definition, not
+        fusible; however, since at least their `strict` parts match, they can at
+        least be topologically reordered.
         """
 
-        def __new__(cls, strict, weak):
+        def __new__(cls, itintervals, guards, syncs, weak):
+            strict = [itintervals, guards, syncs]
```

Comment:
can probably just make i a property returning this three since the object holds those three

Reply:
we could, but it's a nested class and I was trying to keep it as small as possible